### ISSUE TITLE
android back button for leftIcon

### ISSUE THREAD

**Issue by jamesholcomb** [benediktviebahn/react-native-platform-searchbar#11]
Hi,

Trying to render a back button for Android using `leftIcon` and my custom `<Button>` component:

```jsx
          leftIcon={Platform.select({
            android: (
              <Button
                onPress={navigation.goBack}
                iconStyle={{ color: "white" }}
                iconName="arrow-back"
              />
            )
          })}
```

but styles.leftIcon in this code
https://github.com/bviebahn/react-native-platform-searchbar/blob/025c45f81b6dcd7408ac88e448b070407e31321e/src/components/SearchBar/android/index.tsx#L62-L67

seems to be meant for styling an icon and not a passed component.

What do you suggest?

**Comment by bviebahn** [comment]
Hi,
not sure I understand correctly. Your problem is the View wrapping your custom leftIcon component right? I think it needs at least  `marginRight: 'auto'` to look properly, but I'll have to test it. Or maybe there is a better way. I'll have a look at it tomorrow, or you can also submit a PR yourself if you want.

**Comment by jamesholcomb** [comment]
Correct, the View wrapping leftIcon. 

Ultimately trying to achieve this (from react-native-elements)

![31585176-b124cdae-b1bd-11e7-809f-ba966cebc663](https://user-images.githubusercontent.com/17592779/31586716-f6e8ff9c-b1d4-11e7-918f-2a7e11d51b08.gif)

**Comment by bviebahn** [comment]
Can you give some more information what you're having problems with? I don't see the problem with the current styling.

**Comment by jamesholcomb** [comment]
I tried another approach...this works to position the button:

```js
          leftIcon={Platform.select({
            android: (
              <HeaderBackButton  /* from @react-navigation/elements@1.2.1 */
                onPress={this.onDone}
                style={{
                  position: "absolute",
                  marginLeft: -5,
                  marginTop: -5
                }}
              />
            )
          })}
          inputStyle={Platform.select({
            android: {
              marginLeft: 15,
              shadowOpacity: 0,
              shadowRadius: 0,
              shadowColor: "transparent",
              backgroundColor: "transparent"
            }
          })}
```

Without `backgroundColor: "transparent"`, the android ripple will not show.

But now the hitSlop seems to be offset too far left (can only tap the button on the far left side of the icon).

EDIT:  The hitSlop issue may be this: https://github.com/facebook/react-native/issues/27232

**Comment by bviebahn** [comment]
I think you don't need to use `position: "absolute"`, but this probably won't fix your problem. It probably makes sense to remove all styling from the wrapping View except `marginRight: "auto"`.

I was able to reproduce the android ripple problem on react-native 0.63.2 and using Expo: https://snack.expo.dev/wJBeco7qC. Setting a ripple to `borderless: true` causes some weird layout problems. However I couldn't reproduce it in a clean react-native project using 0.66.3, so it looks like this has been fixed. Which version are you on?
